src/history/hs_cursor.c

```diff
@@ -225,6 +224,15 @@ __wt_hs_find_upd(WT_SESSION_IMPL *session, WT_ITEM *key, const char *value_forma
      * history store) to the oldest (earlier in the history store) for a given key.
      */
     read_timestamp = allow_prepare ? txn->prepare_timestamp : txn_shared->read_timestamp;
+
+    /*
+     * A reader without a timestamp should read the largest timestamp in the range, however cursor
+     * search near if given a 0 timestamp will place at the top of the range and hide the records
+     * below it. As such we need to adjust a 0 timestamp to the timestamp max value.
+     */
+    if (read_timestamp == WT_TS_NONE)
```

Comment:
The two changes in this file fix the scenario where we want to search without a timestamp but not with WT_TS_NONE. Most callers to `__wt_hs_cursor_position` do want to search with WT_TS_NONE, only this caller wants to translate WT_TS_NONE to WT_TS_MAX.